analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Invocation.cs

```diff
@@ -41,6 +42,16 @@ public static ProgramState Process(SymbolicContext context, IInvocationOperation
             {
                 state = state.ResetFieldConstraints();
             }
+            state = ProcessThrowHelper(state, invocation);
+            return state;
+        }
+
+        private static ProgramState ProcessThrowHelper(ProgramState state, IInvocationOperationWrapper invocation)
+        {
+            if (KnownMethods.IsDebugFail(invocation.TargetMethod))
+            {
+                return ProgramState.Empty;
```

Comment:
@pavel-mikula-sonarsource  The idea is to suppress any further processing. One option is to reset the state, but this does not stop the processing. A *check* is better suited, right? Except for `Environment.FailFast` (which skips catch, finally, and finalizers), all such methods raise exceptions. Can we simulate that from within a *check*?

Reply:
Yes, InvocationCheck is a better place for now. We will eventually merge those two, but for now, the check it is. There should be something like `return EmptyStates` available. 
See #6118, you might want to base on that as it introduces ability to stop processing.

`Environment.FailFast` is pretty niche. We can just simply stop processing too, as there's nothing we will do about it (and nobody cares about the state of the program afterwards)